skimage/morphology/misc.py

```diff
@@ -124,3 +124,83 @@ def remove_small_objects(ar, min_size=64, connectivity=1, in_place=False):
     out[too_small_mask] = 0
 
     return out
+
+def remove_small_holes(ar, min_size=64, connectivity=1, in_place=False):
+    """Remove connected components smaller than the specified size within a 
+       larger connected object.
+
+    Parameters
+    ----------
+    ar : ndarray (arbitrary shape, int or bool type)
+        The array containing the connected components of interest. If the array
+        type is int, it is assumed that it contains already-labeled objects. The
+        labels are not kept in the output image (this function always outputs
+        a bool image. It is suggested that labeling is completed after using 
+        this function.
+    min_size : int, optional (default: 64)
+        The smallest allowable connected component size.
+    connectivity : int, {1, 2, ..., ar.ndim}, optional (default: 1)
+        The connectivity defining the neighborhood of a pixel.
+    in_place : bool, optional (default: False)
+        If `True`, remove the connected components in the input array itself.
+        Otherwise, make a copy.
+
+    Raises
+    ------
+    TypeError
+        If the input array is of an invalid type, such as float or string.
+    ValueError
+        If the input array contains negative values.
+
+    Returns
+    -------
+    out : ndarray, same shape and type as input `ar`
+        The input array with small holes within connected components removed.
+
+    Examples
+    --------
+    >>> from skimage import morphology
+    >>> a = np.array([[1, 1, 1, 1, 1, 0],
+    ...               [1, 1, 1, 0, 1, 0],
+    ...               [1, 0, 0, 1, 1, 0],
+    ...               [1, 1, 1, 1, 1, 0]], bool)
+    >>> b = morphology.remove_small_holes(a, 2)
+    >>> b
+    array([[True,  True,  True, True, True, False],
+           [True,  True,  True, True, True, False],
+           [True, False, False, True, True, False],
+           [True,  True,  True, True, True, False]], dtype=bool)
+    >>> c = morphology.remove_small_holes(a, 2, connectivity=2)
+    >>> c
+    array([[True,  True,  True,  True, True, False],
+           [True,  True,  True, False, True, False],
+           [True, False, False,  True, True, False],
+           [True,  True,  True,  True, True, False]], dtype=bool)
+    >>> d = morphology.remove_small_holes(a, 2, in_place=True)
+    >>> d is a
+    True
+    """
+    # Should use `issubdtype` for bool below, but there's a bug in numpy 1.7
+    if not (ar.dtype == bool or np.issubdtype(ar.dtype, np.integer)):
```

Comment:
Maybe we can extract this code into a private function and reuse it in both functions?